integration-test-groups/aws2/aws2-sqs-sns/src/main/java/org/apache/camel/quarkus/component/aws2/sqs/it/Aws2SqsSnsResource.java

```diff
@@ -94,10 +135,67 @@ public String sqsReceive() throws Exception {
                 .queueUrls();
     }
 
+    @Path("/sqs/batch")
+    @POST
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.TEXT_PLAIN)
+    public Response sendBatchMessage(List<String> messages) throws Exception {
+        final SendMessageBatchResponse response = producerTemplate.requestBody(
+                componentUri() + "?operation=sendBatchMessage",
+                messages,
+                SendMessageBatchResponse.class);
+        return Response
+                .created(new URI("https://camel.apache.org/"))
+                .entity("" + response.successful().size())
+                .build();
+    }
+
+    @Path("/sqs/delete/message/{queueName}/{receipt}")
+    @DELETE
+    @Produces(MediaType.TEXT_PLAIN)
+    public Response deleteMessage(@PathParam("queueName") String queueName, @PathParam("receipt") String receipt)
+            throws Exception {
+        producerTemplate.sendBodyAndHeader(componentUri(queueName) + "?operation=deleteMessage",
+                null,
+                Sqs2Constants.RECEIPT_HANDLE,
+                receipt);
+        return Response.ok().build();
+    }
+
+    @Path("/sqs/delete/queue/{queueName}")
+    @DELETE
+    @Produces(MediaType.TEXT_PLAIN)
+    public Response deleteQueue(@PathParam("queueName") String queueName) throws Exception {
+        producerTemplate.sendBodyAndHeader(componentUri(queueName) + "?operation=deleteQueue",
+                null,
+                Sqs2Constants.SQS_QUEUE_PREFIX,
+                queueName);
+        return Response.ok().build();
+    }
+
+    @Path("/sqs/queue/autocreate/delayed/{queueName}/{delay}")
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    public List<String> autoCreateDelayedQueue(@PathParam("queueName") String queueName, @PathParam("delay") String delay)
+            throws Exception {
+        // queue creation without any operation resulted in 405 status code
```

Comment:
I'm not sure what's happening there. As far as I can see, [this camel test](https://github.com/apache/camel/blob/main/components/camel-aws/camel-aws2-sqs/src/test/java/org/apache/camel/component/aws2/sqs/integration/SqsComponentLocalstackIT.java) don't need to set operation. Would it be sensible to stick to the camel-aws2-sqs test patterns and maybe separate `autoCreateQueue` and `delayQueue` tests ?

Reply:
Definitely. It was supposed to be as you mention but I must've accidentally drop forgot it

Reply:
Thank you for pointing it out. I must have done something wrong then. I'll try it once more without specifying operation

Reply:
also regarding the separation of the tests -> the easiest way to create a new queue via camel (at least I think it's the easiest) is by using autocreate option. Therefore for creation of `delayQueue` I would yet again use this option. Is there a point in splitting the test in two when `autoCreate` test would be basically also a part of `delayQueue`?

Reply:
Not really, the point of splitting tests would be to stick to camel tests and avoid the 415 status code. Any test configuration covering autoCreate + delayQueue without 415 would be ok. Maybe it's not possible at all ?

Reply:
I'm constantly receiving status code 500 when attempting to create a new route using autoCreate. Even when copying the uri from mentioned camel test.

Reply:
Ok, the main idea is to augment the test coverage. Let's keep the 415 for now.